### PR TITLE
ER-678 Make jobs and usage of client consistent

### PR DESCRIPTION
**Pull request by leewadhams**

### CODE REVIEW

File: src/Jobs/Recruit.Vacancies.Jobs/EmployerDashboardGenerator/EmployerDashboardCreator.cs [hunk 1]
```diff
@@ -5,16 +5,16 @@ namespace Esfa.Recruit.Vacancies.Jobs.EmployerDashboardGenerator
 {
     public class EmployerDashboardCreator
     {
-        private readonly IEmployerDashboardProjectionService _dashboardService;
+        private readonly IEmployerDashboardProjectionService _projectionService;
 
         public EmployerDashboardCreator(IEmployerDashboardProjectionService dashboardService)
```
Comment (sachpatel) [inline src/Jobs/Recruit.Vacancies.Jobs/EmployerDashboardGenerator/EmployerDashboardCreator.cs]: should the parameter be renamed to `projectionService`

File: src/Jobs/Recruit.Vacancies.Jobs/QaDashboard/QaDashboardJob.cs [hunk 2]
```diff
@@ -9,20 +9,20 @@ namespace Esfa.Recruit.Vacancies.Jobs.QaDashboard
     public class QaDashboardJob
     {
         private readonly ILogger<QaDashboardJob> _logger;
-        private readonly IQaDashboardProjectionService _qaDashboardService;
+        private readonly IQaDashboardProjectionService _projectionService;
 
         public QaDashboardJob(ILogger<QaDashboardJob> logger, IQaDashboardProjectionService qaDashboardService)
         {
```
Comment (sachpatel) [inline src/Jobs/Recruit.Vacancies.Jobs/QaDashboard/QaDashboardJob.cs]: maybe here as well.

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/EmployerAccountService.cs [hunk 3]
```diff
@@ -23,9 +23,11 @@ public async Task<IEnumerable<string>> GetEmployerIdentifiersAsync(string userId
         {
             try
             {
-                var accounts = await _accountApiClient.GetUserAccounts(userId);
+                // var accounts = await _accountApiClient.GetUserAccounts(userId);
 
-                return accounts.Select(acc => acc.HashedAccountId);
+                // return accounts.Select(acc => acc.HashedAccountId);
+
+                return new List<string>{"MXD78R"};
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/EmployerAccountService.cs]: Oops!

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/EmployerAccountService.cs [hunk 4]
```diff
@@ -34,19 +36,37 @@ public async Task<IEnumerable<string>> GetEmployerIdentifiersAsync(string userId
...
-                var accounts = await _accountApiClient.GetLegalEntitiesConnectedToAccount(accountId);
+                // var accounts = await _accountApiClient.GetLegalEntitiesConnectedToAccount(accountId);
+
+                // var legalEntitiesTasks = accounts.Select(r => _accountApiClient.GetLegalEntity(accountId, long.Parse(r.Id)));
 
-                var legalEntitiesTasks = accounts.Select(r => _accountApiClient.GetLegalEntity(accountId, long.Parse(r.Id)));
+                // await Task.WhenAll(legalEntitiesTasks.ToArray());
 
-                await Task.WhenAll(legalEntitiesTasks.ToArray());
+                // var entities = legalEntitiesTasks.Select(t => t.Result);
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/EmployerAccountService.cs]: Oops!

File: src/Shared/Recruit.Vacancies.Client/Ioc/ServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -41,60 +41,73 @@ public static void AddRecruitStorageClient(this IServiceCollection services, ICo
         {
             services.Configure<AccountApiConfiguration>(configuration.GetSection("AccountApiConfiguration"));
 
-            RegisterAccountApiClientDeps(services);
-
             services.AddMediatR(typeof(CreateVacancyCommandHandler).Assembly);
             services.AddTransient<IMessaging, MediatrMessaging>();
             
+            // Vacancy Clients
             services.AddTransient<IEmployerVacancyClient, VacancyClient>();
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Vacancies.Client/Ioc/ServiceCollectionExtensions.cs]: Wondering if there is a natural split between Employer and Jobs with regards the client? We have a QaVacancyClient. Is it possible to (and would we want to) have a JobsVacancyClient? One to discuss.

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/EmployerAccountService.cs (hunk 3 above)
Comment (leewadhams) [reply src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/EmployerAccountService.cs]: Yep ooops. Removed that file from commit.

**Pull request merged**